common/debug-interface/src/libra_trace.rs

```diff
@@ -296,3 +317,47 @@ fn abbreviate_crate(name: &str) -> &str {
         _ => name,
     }
 }
+
+// This is exact copy of similar function in log crate
+/// Sets structured logger
+pub fn set_libra_trace(
+    txn_sampling_rate: (u64, u64),
+    block_sampling_rate: (u64, u64),
+) -> Result<(), ()> {
+    unsafe {
+        match LIBRA_TRACE_STATE.compare_and_swap(UNINITIALIZED, INITIALIZING, Ordering::SeqCst) {
+            UNINITIALIZED => {
+                TXN_SAMPLING_RATE = txn_sampling_rate;
+                BLOCK_SAMPLING_RATE = block_sampling_rate;
+                LIBRA_TRACE_STATE.store(INITIALIZED, Ordering::SeqCst);
+                Ok(())
+            }
+            INITIALIZING => {
+                while LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZING {}
+                Err(())
+            }
+            _ => Err(()),
+        }
+    }
+}
+
+/// Checks if libra trace is enabled
+pub fn libra_trace_set() -> bool {
+    LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZED
+}
+
+pub fn is_selected(s: String) -> bool {
+    let v = s.split("::").collect::<Vec<&str>>();
+    if v.len() != 2 && v.len() != 3 {
+        panic!("Invalid node format");
+    }
+    if v[0] == "txn" {
```

Comment:
I think this should be generalized, without hardcoding "txn" in libra trace code.

You can declare SamplingConfig from config crate as following:
`struct SamplingConfig(HashMap<String, String>)`.
Then in libra_trace you can define separate class
```
struct CategorySampling{denominator: u64, nominator: u64}

struct Sampling(HashMap<&'static str, CategorySampling>)
```
You can have a method that will parse SamplingConfig you get from a file into Sampling, and have your global var to hold `Sampling` instance.
Then you don' need separate handling for txn/block, and if someone adds another type of sampling, they can just add it to config as a string.